### ISSUE TITLE
Problems about generate label

### ISSUE THREAD

**Issue by AzulYang** [kamanphoebe/MotionDetection#8]
I want to get labels on my dataset to generate 2D bounding boxes of moving objects, the dataset(KITTI) provides the labels ,the _generate_label.py_ seems only  applicable to the nuScenes dataset ,and it does not applicable to other datasethow ,can I achieve my goal?

**Comment by kamanphoebe** [comment]
Since our scripts are written specifically for the nuScenes dataset, you have to write your own script for manipulating KITTI to gather the necessary information. The labels used here should contain the paths of your .npy files, the coordinates, and the motion status (if aiming for training rather than inference) of the objects . You might check out [`demo_label.csv`](https://github.com/kamanphoebe/MotionDetection/blob/main/demo_label.csv) for details.

**Comment by AzulYang** [comment]
I find it difficult for me to write my own script for manipulating KITTI , cause there is no **ego_pose** and **box_world_location** . At the same time, you mentioned using data without annotation in the experiment part. This part of the article does not cover much. Could you expand on how to get  bounding box positions and ground truths of objects through linear difference.Thanks a lot

**Comment by kamanphoebe** [comment]
In our use case, we aim at objects within a specific distance range, so we calculate the distance through `ego_pose` and `box_world_location`. These two attributes are actually not essential for generating labels if you don't need the distance restriction.   Otherwise, you may find the `objects/location` feature provided by KITTI useful (refer to [here](https://www.tensorflow.org/datasets/catalog/kitti)).

For the generalized experiments, we included the non-keyframes data in nuScenes. As introduced by nuScenes, 
> A keyframe is a frame where the time-stamps of data from all the sensors should be very close to the time-stamp of the sample it points to.

Keyframes are annotated while non-keyframes _between_ keyframes have no annotation. To estimate a non-keyframe's ground truth, the information of its closest previous keyframe is adopted. This is implemented by a few line of codes [here](https://github.com/kamanphoebe/MotionDetection/blob/3cbb80a3bd2cf13c7e145fafa5d92aa5eba59a13/dev/generate_label_expand.py#L164).

**Comment by kamanphoebe** [comment]
Closing the issue due to inactive status. Feel free to re-open if necessary :)
